server/UserRegistration.java

```diff
@@ -0,0 +1,467 @@
+package newbank.server;
+
+import com.sun.tools.javac.Main;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Scanner;
+import java.util.*;
+import java.sql.SQLException;   // If have an error thrown from execution.
+
+public class UserRegistration extends Main {
+
+
+    public static UserRegistration newUser() throws SQLException {
+
+        Scanner sc = new Scanner(System.in);
+        // set all methods to return values - created conversions for sortcode
+        // and account number to make it easier to push through to db.
+
+
+        newUserName nn = new newUserName();
+        String userFullname = nn.GetNameFromUser();
+
+
+        DateChecks dc = new DateChecks();
+        String userDoB = dc.GetDateFromUser();
+
+
+        UserId ui = new UserId();
+        String userUsername = ui.getNewUserId();
+
+
+        newUserAccountNumber acm = new newUserAccountNumber();
+        String userAccountNumber = acm.getNewUserAccountNumber();
+
+
+        newUserSortcode nusort = new newUserSortcode();
+        String userSortCode = nusort.getNewUserSortCode();
+
+
+        NewUserSecurity nus = new NewUserSecurity();
+
+        String userNewPass = nus.inputNewPassword();
+
+
+        NewUserBalance nub = new NewUserBalance();
+
+        Integer userNewBal = Integer.valueOf(nub.balance());
+
+
+        newUserSecretQuestion nsq = new newUserSecretQuestion();
+
+        String userNewSecretQuestion = nsq.secretQuestion();
+
+
+        checkNotARobot cnr = new checkNotARobot();
+
+        boolean NewUserNotARobot = cnr.checkHuman();
+
+        // All already scanned by scanner then passed as values to 'String addNewUser'
+        // The values are then passed to the db.
+
+
+        Connection conn = null;
+
+        try {
+            String url = "jdbc:sqlite:/Users/pmcdowell/Documents/nb/database/bank.sq3";
```

Comment:
This needs to be a relative path, as this absolute path would throw an exception on any other machine other than yours

You can get a relative path in IntelliJ's Database connection tab